pkg/operator/target_config_reconciler.go

```diff
@@ -290,6 +334,47 @@ func (t *TargetConfigReconciler) manageCRD(ctx context.Context, ownerReference m
 	return resourceapply.ApplyCustomResourceDefinitionV1(ctx, t.apiextensionsClient.ApiextensionsV1(), t.eventRecorder, crd)
 }
 
+func (t *TargetConfigReconciler) managePrometheusRole(ctx context.Context, ownerReference metav1.OwnerReference) (*rbacv1.Role, bool, error) {
+	required := resourceread.ReadRoleV1OrDie(bindata.MustAsset("assets/jobset-controller-generated/rbac.authorization.k8s.io_v1_role_jobset-prometheus-k8s.yaml"))
+	required.Namespace = t.operatorNamespace
+	required.OwnerReferences = []metav1.OwnerReference{ownerReference}
+
+	return resourceapply.ApplyRole(ctx, t.kubeClient.RbacV1(), t.eventRecorder, required)
+}
+
+func (t *TargetConfigReconciler) managePrometheusRoleBinding(ctx context.Context, ownerReference metav1.OwnerReference) (*rbacv1.RoleBinding, bool, error) {
+	required := resourceread.ReadRoleBindingV1OrDie(bindata.MustAsset("assets/jobset-controller-generated/rbac.authorization.k8s.io_v1_rolebinding_jobset-prometheus-k8s.yaml"))
+	required.Namespace = t.operatorNamespace
+	required.OwnerReferences = []metav1.OwnerReference{ownerReference}
+
+	for i := range required.Subjects {
+		required.Subjects[i].Namespace = OpenshiftMonitoringNamespace
+	}
+
+	return resourceapply.ApplyRoleBinding(ctx, t.kubeClient.RbacV1(), t.eventRecorder, required)
+}
+
+func (t *TargetConfigReconciler) manageServiceMonitor(ctx context.Context, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, bool, error) {
+	service := resourceread.ReadServiceV1OrDie(bindata.MustAsset("assets/jobset-controller-generated/v1_service_jobset-controller-manager-metrics-service.yaml"))
+	servicemonitor := ReadServiceMonitorV1OrDie(bindata.MustAsset("assets/jobset-controller-generated/monitoring.coreos.com_v1_servicemonitor_jobset-controller-manager-metrics-monitor.yaml"))
+	servicemonitor.Namespace = t.operatorNamespace
+	servicemonitor.OwnerReferences = []metav1.OwnerReference{ownerReference}
+
+	for i, endpoint := range servicemonitor.Spec.Endpoints {
+		endpoint.TLSConfig.ServerName = ptr.To(injectService("METRICS_SERVICE", ptr.Deref(endpoint.TLSConfig.ServerName, ""), service.Name, t.operatorNamespace))
+		// clear out the references
+		endpoint.TLSConfig.SafeTLSConfig.Cert.Secret = nil
```

Comment:
What is the reason of doing this?

Reply:
to remove these fields from the operand manifest since we use the prometheus mounted secret instead